### PR TITLE
Try to use a different action for selecting xcode version

### PR DESCRIPTION
**Pull request by mikepoly**
**Description**
Try to use a different action for selecting xcode version

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,8 +44,12 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3.5.0
-      - name: Switch Xcode Version
-        run: sudo xcode-select -switch "/Applications/Xcode_${{ matrix.xcode_version }}.app"
+      - name: Setup Xcode version
```
Comment (acmaciek) [inline .github/workflows/ci.yml:47]: is lowercase **version** intentional ?

Comment (mikepoly) [reply .github/workflows/ci.yml:47]: This is what they have in their doc 🤷 
Screenshot 2023-11-02 at 3 06 40 PM

Comment (delannoyk) [reply .github/workflows/ci.yml:47]: Yea it doesn't matter, that's just the name

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,8 +44,12 @@ jobs:
...
+        uses: maxim-lobanov/setup-xcode@v1.6.0
+        with:
+          xcode-version: '15.0.0'
+#      - name: Switch Xcode Version
+#        run: sudo xcode-select -switch "/Applications/Xcode_${{ matrix.xcode_version }}.app"
```
Comment (delannoyk) [inline .github/workflows/ci.yml:52]: Can we remove the comments?

Comment (mikepoly) [reply .github/workflows/ci.yml:52]: Still WIP, I wanna see if this works first and then clean up

**Review by acmaciek — APPROVED** [review]
LGTM ✅

**Review by MartinJrP — APPROVED** [review]

**Review by delannoyk — APPROVED** [review]

**Pull request merged**